src/components/pages/paymentPage/ContractPage.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import Layout from '../../../common/layout/Layout';
+import { Button } from 'reactstrap';
+import './paymentPage.scss';
+import { Checkbox } from '../../forms/Fields';
+import { ContractAuthMethods_contractAuthMethods as ContractAuthMethods } from '../../../utils/__generated__/ContractAuthMethods';
+import Form from '../../forms/Form';
+import AuthButton from './authButton/AuthButton';
+
+interface Props {
+  orderNumber: string;
+  contractAuthMethods: ContractAuthMethods[];
+  handleSign: (authMethod: string) => void;
+}
+
+const PaymentPage = ({ contractAuthMethods, orderNumber, handleSign }: Props) => {
+  const { t } = useTranslation();
+  const [termsOpened, setTermsOpened] = useState<boolean>(false);
+  const [termsAccepted, setTermsAccepted] = useState<boolean>(false);
+
+  return (
+    <Layout>
+      <div id="vene-payment-page" className="vene-payment-page">
+        <div className="vene-payment-page__content-container">
+          <div className="vene-payment-page__content">
+            <h2>{t('page.contract.title')}</h2>
+            <p>{t('page.contract.info')}</p>
+            <p className="vene-payment-page__contact-info">
+              {t('page.contract.questions')}&nbsp;
+              <a href="mailto:venepaikat@hel.fi" className="vene-payment-page__link">
+                venepaikat@hel.fi
+              </a>
+            </p>
+          </div>
+        </div>
+        <div className="vene-payment-page__content-container">
+          <div className="vene-payment-page__content vene-payment-page__accept-terms-content">
+            <div>
+              <a
+                href={`${process.env.REACT_APP_API_URL_ROOT}contract_document/${orderNumber}`}
+                className="vene-payment-page__link"
+                rel="noopener noreferrer"
+                target="_blank"
+                onClick={() => setTermsOpened(true)}
+              >
+                {t('page.contract.terms_pdf')}
+              </a>
+            </div>
+
+            <Form
+              onSubmit={(values: { accepted: boolean }) => setTermsAccepted(values.accepted)}
+              initialValues={{}}
+            >
+              {({ values }: { values: { accepted: boolean } }) => (
```

Comment:
I suggest starting with typed ´initialValuesthenvalues` type will be inferred automatically.
```suggestion
              onSubmit={(values) => setTermsAccepted(values.accepted)}
              initialValues={{ accepted: false }}
            >
              {({ values }) => (
```

Reply:
Partially done. Couldn't remove the type definitions from `onSubmit` because it still wasn't able to infer the type.